phoenix-core/src/it/java/org/apache/phoenix/end2end/EmptyColumnIT.java

```diff
@@ -0,0 +1,925 @@
+import org.apache.phoenix.util.TestUtil;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Each column in HBase can have a different timestamp depending on when it gets updated.
+ * The timestamp of the EMPTY_COLUMN can serve as the timestamp of the row (a row in PHOENIX)
+ * and can be used to determine TTL expiration of a row in PHOENIX.
+ *
+ *  These tests are to validate our expectations of the EMPTY_COLUMN behavior.
+ *  Checks :-
+ *  1. Behavior for various tables, views (global and tenant), indexes (global and local)
+ *  2. Behavior under different CF's combinations.
+ *  3. Behavior under different COLUMN ENCODINGS
+ */
+public class EmptyColumnIT extends ParallelStatsDisabledIT {
+
+	private static final Logger logger = LoggerFactory.getLogger(EmptyColumnIT.class);
+	private static int DEFAULT_NUM_ROWS = 5;
+
+	private static class ExpectedTestResults {
+		int numTableRowsExpected;
+		int numTableIndexRowsExpected;
+		int numViewIndexRowsExpected;
+
+		public ExpectedTestResults(int numTableRowsExpected,
+								   int numTableIndexRowsExpected,
+								   int numViewIndexRowsExpected) {
+			this.numTableRowsExpected = numTableRowsExpected;
+			this.numTableIndexRowsExpected = numTableIndexRowsExpected;
+			this.numViewIndexRowsExpected = numViewIndexRowsExpected;
+		}
+	}
+
+    // Scans the HBase rows directly for the empty column and asserts
+	private void assertAllHBaseRowsHaveEmptyColumnCell(byte[] hbaseTableName, byte[] emptyColumnFamilyName, byte[] emptyColumnName, long minTimestamp, int expectedRows) throws IOException, SQLException {
+
+		try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+							 .getTable(hbaseTableName)) {
+
+			Scan allRows = new Scan();
+			allRows.setMaxVersions(1);
+			allRows.setTimeRange(minTimestamp, minTimestamp + Integer.MAX_VALUE);
+			ResultScanner scanner = tbl.getScanner(allRows);
+			int numMatchingRows = 0;
+			for (Result result = scanner.next();  result != null; result = scanner.next()) {
+				numMatchingRows += result.containsColumn(emptyColumnFamilyName, emptyColumnName) ? 1 : 0;
+			}
+
+			assertEquals(String.format(
+					"Expected rows do match for table = %s at timestamp %d",
+					Bytes.toString(hbaseTableName),
+					minTimestamp), expectedRows, numMatchingRows);
+		}
+
+	}
+
+	/**
+	 * -----------------
+	 * Test methods
+	 * -----------------
+	 */
+
+
+
+	@Ignore("fails with java.lang.ArrayIndexOutOfBoundsException: -1 , fails for both Table and GlobalView indexes")
+	// https://issues.apache.org/jira/browse/PHOENIX-5317
+	// https://issues.apache.org/jira/browse/PHOENIX-5322
+	public void testWithBasicTenantViewAndTableIndex() throws Exception {
+
+		// Define the test schema.
+		OtherOptions otherOptions = new OtherOptions();
+		otherOptions.setTableCFs(Lists.newArrayList(null, "A", "B"));
```

Comment:
I don't understand the logic of `Lists.newArrayList(null, "A", "B")` here, what's the format and why we are doing this, thanks.

Reply:
The setTableCFs sets the column family on the table columns. It allows the test writer to specify different column family combinations.